root/static/scripts/common/components/ArtistRoles.js

```diff
@@ -9,30 +9,55 @@
 
 import * as React from 'react';
 
+import hydrate from '../../../../utility/hydrate';
 import commaOnlyList from '../i18n/commaOnlyList';
 import localizeArtistRoles from '../i18n/localizeArtistRoles';
 
+import CollapsibleList from './CollapsibleList';
 import EntityLink from './EntityLink';
 
-type Props = {
-  +relations: $ReadOnlyArray<{
-    +credit: string,
-    +entity: ArtistT,
-    +roles: $ReadOnlyArray<string>,
-  }>,
+type RelationT = {
+  +credit: string,
+  +entity: ArtistT,
+  +roles: $ReadOnlyArray<string>,
 };
 
-const ArtistRoles = ({relations}: Props): React.Element<'ul'> => (
-  <ul>
-    {relations.map(r => (
-      <li key={r.entity.id}>
-        {exp.l('{artist} ({roles})', {
-          artist: <EntityLink content={r.credit} entity={r.entity} />,
-          roles: commaOnlyList(localizeArtistRoles(r.roles)),
-        })}
-      </li>
-    ))}
-  </ul>
+type ArtistRolesProps = {
+  +relations: $ReadOnlyArray<RelationT>,
+};
+
+const buildArtistRoleRow = (relation: RelationT) => {
+  return (
+    <li key={relation.entity.id + '-' + relation.credit}>
```

Comment:
Credit needed to be added to the key to avoid clashes when the same artist appeared under two different credits.